docs/source/howto/interact.rst

```diff
@@ -0,0 +1,112 @@
+    Out[2]: <aiida.manage.configuration.profile.Profile at 0x7fccfd6c50a0>
+
+One can pass a particular profile name to :meth:`~aiida.load_profile`, otherwise the default profile is loaded.
+Alternatively, the command line interface ``verdi`` provides a handy shortcut:
+
+.. code-block:: console
+
+    $ verdi shell
+
+This command will open a normal IPython shell but automatically loads the AiiDA profile.
+In addition, certain modules from AiiDA's API that are used very often are automatically imported.
+
+
+.. _how-to:interact-notebook:
+
+Interactive notebooks
+=====================
+
+Similar to :ref:`interactive shells <how-to:interact-shell>`, AiiDA is also directly compatbile with interactive Python notebooks, such as `Jupyter <https://jupyter.org/>`_.
+To use AiiDA's Python API in a notebook, first a profile has to be loaded:
+
+.. code-block:: ipython
+
+    In [1]: from aiida import load_profile
+
+    In [2]: load_profile()
+    Out[2]: <aiida.manage.configuration.profile.Profile at 0x7fccfd6c50a0>
+
+One can pass a particular profile name to :meth:`~aiida.load_profile`, otherwise the default profile is loaded.
+
+
+.. _how-to:interact-scripts:
+
+Scripts
+=======
+
+AiiDA's Python API can be used in Python scripts mixed with any other Python code.
+The only requirement, is that before the API is used, an AiiDA profile is loaded.
+This can be accomplished by adding the following lines to the script:
+
+.. code-block:: python
+
+    from aiida import load_profile
+    load_profile()
+
+One can pass a particular profile name to :meth:`~aiida.load_profile`, otherwise the default profile is loaded.
+Alternatively, instead of adapting the script with these lines, one can call it through the command line interface:
+
+.. code-block:: console
+
+    verdi run script.py
+
+The ``verdi`` CLI will automatically load the default profile, before calling the actual script, passing any command line arguments that may have been specified.
+Note that a :ref:`different profile can be selected <topics:cli:profile>` using the ``--profile`` option, just as for all other ``verdi`` commands.
+
+Finally, one can also add AiiDA's specific `shebang <https://en.wikipedia.org/wiki/Shebang_(Unix)>`_ to the top of the file.
+
+.. code-block:: bash
+
+    #!/usr/bin/env runaiida
+
+When a script starts with this shebang, when it is executed, it is automatically passed to ``verdi run`` just as if it would have been called through ``verdi run`` directly.
+This has the advantage that one no longer has to explicitly type ``verdi run`` when running the script, but can simply make it executable and execute it directly.
+The downside is that it does not allow to specify a particular profile, but it always loads the default profile.
+
+
+.. _how-to:interact-restapi:
+
+REST API
+========
+
+AiiDA ships with a built in REST API, that allows you to query the data of a particular profile.
+Refer to section TODO_LINK to learn how to start the REST API.
```

Comment:
You already mentioned this, but I'm adding as a review comment so we don't forget - this needs to be updated as soon as the REST API docs go back in place.